### PR TITLE
Use globally shared HTMLRender

### PR DESCRIPTION
**Pull request by wxiaoguang**
The old `HTMLRender` is not ideal.

1. It shouldn't be initialized multiple times, it consumes a lot of memory and is slow.
2. It shouldn't depend on short-lived requests, the `WatchLocalChanges` needs a long-running context.
3. It doesn't make sense to use FuncsMap slice.

HTMLRender was designed to only work for GItea's specialized 400+ templates, so it's good to make it a global shared instance.

### CODE REVIEW

File: modules/templates/mailer.go [hunk 1]
```diff
@@ -14,6 +15,8 @@ import (
 	"code.gitea.io/gitea/modules/setting"
 )
 
+var mailSubjectSplit = regexp.MustCompile(`(?m)^-{3,}\s*$`)
```
Comment (6543) [inline modules/templates/mailer.go:18]:
```suggestion
var mailSubjectSplit = regexp.MustCompile(`(?m)^-{3,}[\s]*$`)
```
why was it changed?

File: modules/templates/htmlrenderer.go [hunk 2]
```diff
@@ -86,35 +88,36 @@ func (h *HTMLRender) CompileTemplates() error {
 	return nil
 }
 
-// HTMLRenderer returns the current html renderer for the context or creates and stores one within the context for future use
-func HTMLRenderer(ctx context.Context) (context.Context, *HTMLRender) {
-	if renderer, ok := ctx.Value(rendererKey).(*HTMLRender); ok {
-		return ctx, renderer
-	}
+// HTMLRenderer returns the globally shared html renderer
```
Comment (6543) [inline modules/templates/htmlrenderer.go]:
```suggestion
// HTMLRenderer init once and returns the globally shared html renderer
```

File: modules/templates/mailer.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/templates/mailer.go:18]: Auto simplified by my IDE.

`\s` just equals to `[\s]`, and my IDE doesn't show a warning for it.

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
IMO in most cases it doesn't affect performance, so no need to mark it as performance or CPU. Only some rare cases that the ctx doesn't contain the pre-created render would cause multiple instances.

**Review by 6543 — APPROVED** [review]

**Pull request merged**